enhancements/workload-partitioning/rationalizing-configuration.md

```diff
@@ -0,0 +1,478 @@
+   decide when to mutate pods.
+7. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+#### Day 2: Add a New Node to an Existing MachineConfigPool
+
+In a cluster with partitioning enabled and fully configured, adding a
+new node to an existing MachineConfigPool is safe because the
+MachineConfig will set up kubelet and CRI-O on the host correctly.
+
+1. The MCO applies the MachineConfigs to the node.
+2. Kubelet starts and sees the config file enabling partitioning. It
+   advertises the workload resource type on the Node and mutates
+   static pods with partitioning annotations.
+3. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+#### Day 2: Add a New MachineConfigPool
+
+Adding a new MachineConfigPool is safe only if the MachineConfigPool
+has a PerformanceProfile associated and the PAO is installed.
+
+1. PAO reads PerformanceProfile to generate new MachineConfig for the
+   pool with CRI-O and `kubelet` settings.
+2. MachineConfigs are applied to appropriate nodes, rebooting them in
+   the process.
+3. Kubelet starts and sees the config file enabling partitioning. It
+   advertises the workload resource type on the Node and mutates
+   static pods with partitioning annotations.
+4. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+If the MachineConfigPool does not match a PerformanceProfile, there
+will be no cpuset information and the PAO will generate MachineConfigs
+with partitioning enabled but not tied to a cpuset. See the [Risks
+section](#risks-and-mitigations) for details.
+
+If the PAO is not installed, it will not generate the overriding
+MachineConfigs for the new MachineConfigPool. See the [Risks
+section](#risks-and-mitigations) for details.
+
+### Risks and Mitigations
+
+There is some risk of shipping a feature with part of the
+implementation in the OpenShift release payload but the enabling tool
+delivered separately. PAO is considered to be a "core" OpenShift
+component, even though it is delivered separately. There is not a
+separate SKU for it, for example. The PAO team has been working
+closely with the node team on the implementation of this feature, so
+we do not anticipate any issues delivering the finished work in this
+way.
+
+If a MachineConfigPool exists without a matching PerformanceProfile,
+there will be no cpuset information and the PAO will generate
+MachineConfigs with partitioning enabled but not tied to a cpuset.
+This is somewhat safe, but may lead to unexpected behavior. The
+mutated pods would float across the full cpuset in the same way as if
+partitioning was not enabled and the cpushares would not be deducted
+from available CPUs, potentially leading to over commit scenarios.
+
+If the PAO is not installed into a cluster with workload partitioning
+enabled and configured, adding a new MachineConfigPool can be unsafe.
```

Comment:
Like above, I think we should carefully consider custom pool use cases and whether we want to support them at all.

Reply:
I don't think we're going to be able to avoid supporting custom pools.